R/spaces.R

```diff
@@ -0,0 +1,137 @@
+spaces_base <- "nyc3.digitaloceanspaces.com"
+
+check_space_access <- function(spaces_key) {
+  tmp <- if (is.null(spaces_key)) Sys.getenv("DO_SPACES_ACCESS_KEY") else spaces_key
+  if (tmp == "") stop("Need a digital ocean spaces access key defined in your session", call. = FALSE) else tmp
+}
+
+check_space_secret <- function(spaces_secret) {
+  tmp <- if (is.null(spaces_secret)) Sys.getenv("DO_SPACES_SECRET_KEY") else spaces_secret
+  if (tmp == "") stop("Need a digital ocean spaces access key defined in your session", call. = FALSE) else tmp
+}
+
+#' @param x Object to coerce to a space
+#' @export
+#' @rdname spaces
+as.space <- function(x) UseMethod("as.space")
+#' @export
+as.space.space <- function(x) x
+
+#' @export
+print.space <- function(x, ...) {
+  cat("<space>", x$Name, "\n", sep = "")
+  cat("  Created at: ", x$CreationDate, "\n")
+}
+
+#' @export
+summary.space <- function(object, ...) {
+  space_info <- space_info(name = object$Name, ...)
+
+  # obtain total size used by space
+  size <- space_size(space_info)
+
+  # obtain number of files in space
+  n_files <- space_files(space_info)
+
+  cat("<space_detail>", object$Name, "\n", sep = "")
+  cat("  Size (GB):    ", size, "\n")
+  cat("  Files:        ", n_files, "\n", sep = "")
+  cat("  Created at:   ", object$CreationDate, "\n")
+}
+
+#' Spaces storage operations
+#'
+#' \describe{
+#'  \item{space}{List space contents}
+#'  \item{spaces}{List spaces in your digital ocean account}
+#'  \item{space_create}{Create a new space}
+#' }
+#'
+#' @param name (character) Space name.
+#' @param spaces_key (character) String containing a spaces access key. If
+#'   missing, defaults to value stored in an environment variable \code{DO_SPACES_ACCESS_KEY}.
+#' @param spaces_secret (character) String containing the secret associated
+#'   with the spaces key. If missing, defaults to value stored in an environment
+#'   variable \code{DO_SPACES_SECRET_KEY}.
+#' @param ... Additional arguments passed down to \code{\link[aws.s3]{bucketlist}},
+#'   \code{\link[aws.s3]{get_bucket}}, \code{\link[aws.s3]{put_bucket}} functions.
+
+#' @importFrom aws.s3 bucketlist
+#' @export
+#' @rdname spaces
+spaces_GET <- function(spaces_key = NULL, spaces_secret = NULL, ...) {
```

Comment:
This was also supposed to be an internal function, since the `spaces()` function is the one intended to be user facing.  I'll make sure this is internal in my revisions.

Reply:
ok, thx